File: logger.py
"""
Logging functionality for tracker
"""

import logging
import os
from datetime import datetime


class TrackerLogger:
    def __init__(self, log_dir='logs'):
        self.log_dir = log_dir
        self.setup_logger()

    def setup_logger(self):
        """Setup logging configuration"""
        if not os.path.exists(self.log_dir):
            os.makedirs(self.log_dir)

        log_file = os.path.join(
            self.log_dir,
            f"tracker_{datetime.now().strftime('%Y%m%d')}.log"
        )

        # Clear any existing handlers to prevent duplicate logging
        logger = logging.getLogger('SystemTracker')
        logger.handlers.clear()

        # Create handlers
        file_handler = logging.FileHandler(log_file)
        stream_handler = logging.StreamHandler()

        # Create formatter and add it to handlers
        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
        file_handler.setFormatter(formatter)
        stream_handler.setFormatter(formatter)

        # Add handlers to logger
        logger.addHandler(file_handler)
        logger.addHandler(stream_handler)
        logger.setLevel(logging.INFO)

        self.logger = logger

    def log_stats(self, stats_type, stats_data):
        """Log system statistics"""
        self.logger.info(f"{stats_type}: {stats_data}")

    def log_alert(self, message):
        """Log alert messages"""
        self.logger.warning(f"ALERT: {message}")

    def log_error(self, error_message):
        """Log error messages"""
        self.logger.error(f"ERROR: {error_message}")

File: src/logger.rs
use chrono::Local;
use std::fs::{self, OpenOptions};
use std::io::Write;
use std::path::PathBuf;

pub struct TrackerLogger {
    log_dir: PathBuf,
}

impl TrackerLogger {
    pub fn new(log_dir: &str) -> Self {
        let log_dir = PathBuf::from(log_dir);
        if !log_dir.exists() {
            fs::create_dir_all(&log_dir).ok();
        }
        TrackerLogger { log_dir }
    }

    fn get_log_file(&self) -> PathBuf {
        let date = Local::now().format("%Y%m%d").to_string();
        self.log_dir.join(format!("tracker_{}.log", date))
    }

    fn write_log(&self, level: &str, message: &str) {
        let timestamp = Local::now().format("%Y-%m-%d %H:%M:%S").to_string();
        let log_message = format!("{} - SystemTracker - {} - {}\n", timestamp, level, message);

        if let Ok(mut file) = OpenOptions::new()
            .create(true)
            .append(true)
            .open(self.get_log_file())
        {
            let _ = file.write_all(log_message.as_bytes());
        }
    }

    pub fn log_stats(&self, stats_type: &str, stats_data: &str) {
        let message = format!("{}: {}", stats_type, stats_data);
        self.write_log("INFO", &message);
    }

    pub fn log_alert(&self, message: &str) {
        let alert_message = format!("ALERT: {}", message);
        self.write_log("WARNING", &alert_message);
    }

    #[allow(dead_code)]
    pub fn log_error(&self, error_message: &str) {
        let error_msg = format!("ERROR: {}", error_message);
        self.write_log("ERROR", &error_msg);
    }
}

impl Default for TrackerLogger {
    fn default() -> Self {
        Self::new("logs")
    }
}
